.github/ISSUE_TEMPLATE/new-release-template.md

```diff
@@ -7,30 +7,41 @@ assignees: comcmaho, amdeel, davidmrdavid, bachuv
 
 ---
 
-**Prep Release: (assigned to:)**
-_Due: <3-business-days-before-release>_
-- [ ] Merge all features and fixes into the `dev` branch
-- [ ] Publish updated versions of DurableTask.Core and DurableTask.AzureStorage to app-service MyGet feed
-- [ ] Update dependencies and increment extension version in the `dev` branch.
-- [ ] Publish signed version of Microsoft.Azure.WebJobs.Extensions.DurableTask to app-service MyGet feed
-- [ ] Close out or punt remaining GitHub issues for the current milestone
+**Prep DTFx Release: (assigned to:)**
+_Due: <2-3-business-days-before-release>_
+- [ ] Increment updated DTFx packages (either DT-AzureStorage only or DT-AzureStorage, DT-Core, and DT-Redis)
+- [ ] Delete DTFx test packages from myget.
+- [ ] Update OneBranch durabletask repo, run signing pipeline (using these notes:  [Durable Task Framework Build Pipeline] (https://microsoft.sharepoint.com/teams/AzureWebjobs/_layouts/OneNote.aspx?id=%2Fteams%2FAzureWebjobs%2FSiteAssets%2FAzureWebjobs%20Notebook&wd=target%28Planning%2FFunctions%20Post-GA.one%7CA43CF112-7272-481A-B23E-9AA5CA8EEE06%2FDurable%20Task%20Framework%20Build%20Pipeline%7CD0946823-6FB0-44E3-A57F-E252617B69CD%2F%29
+onenote:https://microsoft.sharepoint.com/teams/AzureWebjobs/SiteAssets/AzureWebjobs%20Notebook/Planning/Functions%20Post-GA.one#Durable%20Task%20Framework%20Build%20Pipeline&section-id={A43CF112-7272-481A-B23E-9AA5CA8EEE06}&page-id={D0946823-6FB0-44E3-A57F-E252617B69CD}&end) ).
+- [ ] Publish DTFx packages to myget for testing. (either DT-AzureStorage only or DT-AzureStorage, DT-Core, DT-Redis, and DT-Emulator)
 
-
-**Validation (assigned to: )**
+**Prep Release (assigned to: )**
 _Due: <2-business-days-before-release>_
-- [ ] Run private performance tests and ensure no regressions
-- [ ] Smoke test Functions V1 and Functions V3 .NET apps
-- [ ] Smoke test JavaScript and Python apps
-- [ ] Check for package size, make sure it's not surprisingly heavier than a previous release
+- [ ] Update Durable Functions references (Analyzer? DTFx?) and increment version.
```

Comment:
Another thing we could consider is incrementing the version at the *start* of every milestone. This would make it easier to do nightly releases that have sensible version numbers and is more consistent with build systems I've seen used around the company.

Reply:
We could accomplish that by moving this step until after we merge dev to main.

Reply:
This is for another thread, but I want to discuss nightly releases further, I don't really understand them, how that would work with DTFx and the extension, and if they would be beneficial.

Reply:
@amdeel, the goal of nightly releases would be that as soon as we merge a fix, there would be a build available for customers to get access to that fix in an unofficial capacity without us uploading the build to myget manually. It makes it much easier for customers to validate fixes before our official releases.

We could eventually look at having test bot run our smoke/load tests against the nightly release bits so we identify regressions earlier in the release process as well.

Reply:
My only concern is that eagerly incrementing the version number might complicate making major-version releases. For example, what happens if the PR that justified the major-version upgrade isn't complete by the milestone's end date? Do we delay the entire release? Or do we revert the major-version upgrade and merge a minor-version upgrade instead to unblock the release?

Reply:
I don't think we would commit to a major version release if we weren't sure about it. The issue would more likely be that we increment to 2.4.2 when we never intend to release 2.4.2. Taking nightly releases out of the conversation, we can increment the version again to 2.5.0 when we commit on having that as our next release.

Reply:
That sounds reasonable to me. Most of the time we have a _fairly_ good idea if we are doing a patch or minor release next, and there is minimal overhead to change it if we change our mind a couple weeks later.